### PR TITLE
Add NXP.PCF8563 impl, and fix NXP.PCA8565

### PR DESCRIPTION
**Pull request by pojiro**
Currentry I'm trying to use [EdgeBox RPI 200](https://jp.seeedstudio.com/EdgeBox-RPi-200-CM4102008-p-5487.html) as Nerves. It has NXP.PCF8563 RTC whose spec is almost same as NXP.PCA8565.

So I try to use the RTC with nerves_time_rtc_nxp, this PR achives it. Below is the capture, it works.

image

### CODE REVIEW

File: lib/nerves_time/rtc/nxp/pca8565.ex [hunk 1]
```diff
@@ -93,6 +97,12 @@ defmodule NervesTime.RTC.NXP.PCA8565 do
     minute_bcd = from_integer(date_time.minute)
     hour_bcd = from_integer(date_time.hour)
     day_bcd = from_integer(date_time.day)
+
+    weekday_bcd =
+      Calendar.ISO.day_of_week(date_time.year, date_time.month, date_time.day, :sunday)
+      |> then(fn {day_of_week, 1, 7} -> day_of_week - 1 end)
+      |> from_integer()
+
```
Comment (pojiro) [inline lib/nerves_time/rtc/nxp/pca8565.ex:105]: [Calendar.ISO.day_of_week](https://hexdocs.pm/elixir/1.14.5/Calendar.ISO.html#day_of_week/4) returns `1..7` for weekdays.

So `day_of_week`, 1..7 should be decriment 1 for devive.

PCA8565, PCF8563 both uses `0..6` for weekdays, 0 is sunday for both device.

refs. 
https://www.nxp.com/docs/en/data-sheet/PCA8565.pdf, sec. 8.4.5
https://www.nxp.com/docs/en/data-sheet/PCF8563.pdf, sec. 8.4.5

File: lib/nerves_time/rtc/nxp/pca8565.ex [hunk 2]
```diff
@@ -54,11 +53,16 @@ defmodule NervesTime.RTC.NXP.PCA8565 do
...
   end
 
   @impl NervesTime.RealTimeClock
   def get_time(state) do
-    get_time_from_rtc(state)
+    with {:ok, time} <- get_time_from_rtc(state) do
+      {:ok, time, state}
+    else
+      _ -> {:unset, state}
+    end
```
Comment (pojiro) [inline lib/nerves_time/rtc/nxp/pca8565.ex:65]: `set_time` and `get_time` returns should be match behaviour I/F, so I fixed.

**Comment by pojiro** [comment]
@fhunleth @ConnorRigby Could you review this when you are free?

**Comment by fhunleth** [comment]
Hi @pojiro - Thanks for the PR and for at-ing me since I didn't get the notification. I don't have this RTC. Let me see if Connor is still using it. Would you mind if I gave you access to this repository to help maintain it? You might be using it more than us right now.

**Review by ConnorRigby — CHANGES_REQUESTED** [review]
I don't have hardware to test it unfortunately, but this is a very good looking PR. I checked the linked datasheets and it looks correct to me on face value. It feels good to merge to me. Would you mind squashing the commits?

**Comment by pojiro** [comment]
@fhunleth @ConnorRigby Thanks for quick reply.

> Would you mind if I gave you access to this repository to help maintain it?

> Would you mind squashing the commits?

Both, I don't mind. Thanks to give me a chance to contribute :)

**Comment by pojiro** [comment]
@ConnorRigby I squashed commits to one.

**Review by ConnorRigby — APPROVED** [review]

**Pull request merged**